client/lib/signup/private-by-default.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import debugFactory from 'debug';
+
+const debug = debugFactory( 'calypso:signup:private-by-default' );
+
+interface PrivateByDefaultSiteSettings {
+	flowName?: string;
+}
+
+/**
+ * Should the site be private by default
+ * @param siteSettings PrivateByDefaultSiteSettings
+ * @returns `true` for private by default & `false` for not
+ */
+export function shouldBePrivateByDefault( {
+	flowName = '',
+}: Readonly<PrivateByDefaultSiteSettings> ): boolean {
+	if ( flowName.match( /^ecommerce/ ) ) {
+		// ecommerce plans go atomic after checkout. These sites should default to public for now.
+		return false;
+	}
+
+	return true;
+}
+
+/**
+ * Get the numeric value that should be provided to the "new site" endpoint
+ *
+ * @param dependencies The `dependencies` passed to the `apiRequestFunction` step action function call
+ * @param stepData The `stepData` passed to the `apiRequestFunction` step action function call
+ * @returns `-1` for private by default & `1` for public
+ */
+export function getNewSitePublicSetting( dependencies: object, stepData: object ): number {
```

Comment:
_Observation_

`object` is very open. Typing these interfaces would likely be beneficial across signup in the future.

Reply:
I was thinking about how it might be good to have some placeholder signup types that don't check much for now but give us a place to add things as we gradually add types e.g.

```typescript
interface Dependencies {
  // things we've typed that will give us auto-complete and type checking
  someDependency?: string;

  // catch-all for things we haven't typed yet
  [other: string]: any;
}
```

Probably for another PR though 😅

Reply:
I was playing with a similar idea. I think it will be important for gradual typing to differentiate between `any` and _not yet typed for some reason_:

https://github.com/Automattic/wp-calypso/blob/6b7f40bd4f503090c427171b1eae1a1930f2b15c/client/types.ts#L37-L44

(☝️ that's not available yet, just a sketching so far)

Reply:
I like this!